metrics-operator/controllers/common/providers/prometheus/prometheus.go

```diff
@@ -67,3 +67,45 @@ func (r *KeptnPrometheusProvider) EvaluateQuery(ctx context.Context, metric metr
 	}
 	return value, b, nil
 }
+
+func EvaluateQueryWithRange(ctx context.Context, metric metricsapi.KeptnMetric, r *KeptnPrometheusProvider, api prometheus.API) (model.Value, prometheus.Warnings, error) {
+	queryTime := time.Now().UTC()
+	// Get the duration
+	queryInterval, err := time.ParseDuration(metric.Spec.Range.Interval)
+	if err != nil {
+		return nil, nil, err
+	}
+	// Convert type Duration to type Time
+	startTime := queryTime.Add(-queryInterval).UTC()
+	r.Log.Info("Running query: /api/v1/query_range?query=" + metric.Spec.Query + "&start=" + startTime.String() + "&end=" + queryTime.String())
```

Comment:
let's use the unix timestamp instead of the string representation of the timestamps

Reply:
So should I convert the unix timestamp to string afterward like `strconv.FormatInt(startTime.Unix(),10)` because there's an error if we just passing the Unix timestamp over here
```
invalid operation: "Running query: /api/v1/query_range?query=" + metric.Spec.Query + "&start=" + startTime.Unix() (mismatched types string and int64)
```

Reply:
ideally, we should log the query exactly as we send it to the api - To prevent this error, I would say using `fmt.Sprintf` might be the cleanest approach:

```
r.Log.Info(fmt.Sprintf(
		"Running query: /api/v1/query_range?query=%s&start=%d&end=%d",
		metric.Spec.Query,
		startTime.Unix(), queryTime.Unix(),
	))
```